File: ed/go/examples/benchmark.1.go
package main

import (
    "encoding/json"
    "net/http"
    "strings"
)

type Response struct {
    Id string
    Msg string
}

func main() {
    http.HandleFunc("/v1/id/", func(w http.ResponseWriter, r *http.Request) {
        if r.Method != "GET" {
            w.Write([]byte(`{error:"501 Not Implemented"}`))
            return
        }

        url := r.URL.Path
        p := strings.LastIndex(url, "/") + 1
        id := url[p:]

        res := Response{Id: id, Msg: "OK."}
        e := json.NewEncoder(w)
        e.Encode(res)
    })
    http.ListenAndServe(":8080", nil)
}

File: ed/nodejs/examples/benchmark.1.js
const http = require('http');

http.createServer((req, res) => {
    if (req.url.startsWith('/v1/id/')) {
        if (req.method != 'GET') {
            res.end('{error:"501 Not Implemented"}');
            return;
        }

        const url = req.url;
        const p = url.lastIndexOf('/') + 1;
        const id = url.substring(p, url.length)

        res.end(JSON.stringify({Id: id, Msg: "OK."}));
    }
}).listen(8080, '0.0.0.0');
